### PR TITLE
Provide ArrayStack.Clone method

### PR DESCRIPTION
**Pull request by TheByKotik**
Provide ArrayStack.Clone method.

Tested with [this script](https://gist.github.com/TheByKotik/d84d2fbe80581df600def7d2e1b4477d).

### CODE REVIEW

File: plugins/include/adt_stack.inc [hunk 1]
```diff
@@ -136,6 +144,17 @@ methodmap ArrayStack < Handle
...
+/**
+ * Clones an stack, returning a new handle with the same size and data. This should NOT
+ * be confused with CloneHandle. This is a completely new handle with the same data but
+ * no relation to the original. You MUST close it.
+ *
+ * @param stack         Stack handle to be cloned
+ * @return              New handle to the cloned stack object
+ * @error               Invalid Handle
+ */
+native ArrayStack CloneStack(Handle stack);
```
Comment (dvander) [inline plugins/include/adt_stack.inc:156]: No reason to provide old-style natives anymore.

Comment (TheByKotik) [reply plugins/include/adt_stack.inc:156]: Then i remove it?

Comment (dvander) [reply plugins/include/adt_stack.inc:156]: Yeah, indeed. This encourages using the newer APIs.

**Comment by KyleSanderson** [comment]
thanks for the contribution @TheByKotik

File: core/logic/smn_adt_stack.cpp [hunk 2]
```diff
@@ -387,6 +413,7 @@ static cell_t GetStackBlockSize(IPluginContext *pContext, const cell_t *params)
 REGISTER_NATIVES(cellStackNatives)
 {
 	{"CreateStack",					CreateStack},
+	{"CloneStack",					CloneStack},
```
Comment (rumblefrog) [inline core/logic/smn_adt_stack.cpp:416]: Would this need cleanup as well?

Comment (KyleSanderson) [reply core/logic/smn_adt_stack.cpp:416]: eh, it's an internal thing. We're all adults, if someone really wants to get at this they can.

Comment (TheByKotik) [reply core/logic/smn_adt_stack.cpp:416]: Maybe then create RP with total old-style cleanup?
Or it's excess?

**Comment by peace-maker** [comment]
This is a duplicate of #852

**Pull request merged**